// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package hindsightgrpc;

// The greeting service definition.
service HindsightGRPC {
  // Sends a greeting
  rpc Exec (ExecRequest) returns (ExecReply) {}
}

message HindsightContext {
  int64 trace_id = 1;
  int64 span_id = 2;
  bool triggerflag = 3;
  repeated string breadcrumb = 4;
}

message OtelContext {
  string trace_id = 1;
  string span_id = 2;
  bool sample = 3;
}

message ExecRequest {
  string api = 1;
  int64 interval = 2;
  bool debug = 3;
  string payload = 4;
  HindsightContext hindsight = 5;
  OtelContext otel = 6;
}

message ExecReply {
  string payload = 1;
  HindsightContext hindsight = 2;
}
